mobile/src/home-view.js

```diff
@@ -269,13 +270,17 @@ class HomeView extends Component {
           return a.order - b.order
         })
         this.originalOrder(otherQuestions)
-        var orderedQuestions = pinnedQuestions.concat(otherQuestions)
-        if (currentSort === "Approved" || "Popular" || "Recent" && orderedQuestions.length) { 
-          orderedQuestions = orderedQuestions.filter(item => item.block === false && item.answered === false && item.approve && item.new === false && item.session === session.key)
+        const allQuestions = pinnedQuestions.concat(otherQuestions)
+        let orderedQuestions = allQuestions
+        switch (currentSort) {
+          case "Answered" : orderedQuestions = allQuestions.filter(item => item.answered && item.session === session.key)
+          break;
+          case "Blocked" : orderedQuestions = allQuestions.filter(item => item.block && item.new === false && item.session === session.key) 
+            break;
+          case "New" : orderedQuestions = allQuestions.filter(item => item.approve === false && item.new && item.session === session.key)
+            break;
+          default : orderedQuestions = allQuestions.filter(item => item.block === false && item.answered === false && item.approve && item.new === false && item.session === session.key)
```

Comment:
It is still customary to put a `break` after a default case.  Also, it is normal to put all statements following a `case` on their own lines, e.g.

```javascript
case "Answered":
  orderedQuestions = allQuestions.filter(item => item.answered && item.session === session.key)
  break
```

Also make sure all of the statements and `break`s are indented in the same way here.